src/usr/local/buildpack/utils/environment.sh

```diff
@@ -72,24 +70,27 @@ function get_tool_version_env () {
 
 function setup_env_files () {
   # env helper, loads tool specific env
+  local install_dir
+  install_dir=$(get_install_dir)
+
   cat >> "$ENV_FILE" <<- EOM
-export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}" USER_HOME="/home/${USER_NAME}"
+export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}" USER_HOME="${USER_HOME}"
 
 # openshift override unknown user home
 if [ "\${EUID}" != 0 ]; then
   export HOME="\${USER_HOME}"
 fi
 
-if [ -d /usr/local/env.d ]; then
-  for i in /usr/local/env.d/*.sh; do
+if [ -d "${install_dir}/env.d" ]; then
+  for i in "${install_dir}/env.d/"*.sh; do
     if [ -r \$i ]; then
       . \$i
     fi
   done
   unset i
 fi
-if [ -d /home/"${USER_NAME}"/env.d ]; then
-  for i in /home/"${USER_NAME}"/env.d/*.sh; do
+if [ -d "${USER_HOME}"/env.d ]; then
+  for i in "${USER_HOME}"/env.d/*.sh; do
```

Comment:
```suggestion
if [ -d "${USER_HOME}/env.d" ]; then
  for i in "${USER_HOME}/env.d/"*.sh; do
```

Would  `"${USER_HOME}/env.d/*.sh"` work too? 🤔

Reply:
It would not, because then the glob pattern would not work: https://github.com/koalaman/shellcheck/wiki/SC2086

```
When quoting composite arguments, make sure to exclude globs and brace expansions, which lose their special meaning in double quotes: "$HOME/$dir/src/*.c" will not expand, but "$HOME/$dir/src"/*.c will.
```

Reply:
OK, updated suggestion. 🙃